include/boost/beast/http/impl/read.ipp

```diff
@@ -34,17 +35,30 @@ namespace detail {
 
 //------------------------------------------------------------------------------
 
+template<typename DynamicBuffer>
+boost::optional<typename DynamicBuffer::mutable_buffers_type>
+prepare_or_error(DynamicBuffer& buffer, error_code& ec)
```

Comment:
I like this very much, this would be helpful to use everywhere. It should probably be available as a `detail` function, maybe in `<boost/core/detail/type_traits.hpp>`